packages/components/buttons/icon-button/src/icon-button.tsx

```diff
@@ -59,7 +59,7 @@ export type TIconButtonProps<
     event: MouseEvent<HTMLButtonElement> | KeyboardEvent<HTMLButtonElement>
   ) => void;
   /**
-   * The container shape of the button.
+   * @deprecated
```

Comment:
Wondering if we should mark it as deprecated only when we turn on the new design. In any case, it's useful to add a message to inform the user what to do (or what the consequence is).

Reply:
I meant to add the annotation only after we turn on the new design.

Let's keep it simple and keep the annotation but provide a useful message to the user.

Reply:
Oh, yes I think I got it, thanks. Would that work for you?
<img width="568" alt="Screenshot 2023-02-28 at 10 49 37" src="https://user-images.githubusercontent.com/49066275/221816439-3692e33d-cdb2-46c8-b081-881effa14d6c.png">
```ts
/**
 * @deprecated This prop is only used in the old theme. For the new theme this prop will not be taken into account and `square` is used by default
 */
```

Reply:
I like that. Seems like we're otherwise overdoing it a bit.